src/storage/src/mut_records.rs

```diff
@@ -136,7 +178,92 @@ impl MutFileRecords {
 
     #[allow(unused)]
     pub async fn flush(&mut self) -> Result<(), IoError> {
-        self.f_sink.flush().await
+        self.flush_count.fetch_add(1, Ordering::Relaxed);
+        let mut f_sink = self.f_sink.lock().await;
+        debug!("flush: count {:?}", self.flush_count);
+        f_sink.flush().await
+    }
+
+    #[allow(unused)]
+    pub fn flush_count(&self) -> u32 {
+        self.flush_count.load(Ordering::Relaxed)
+    }
+
+    async fn delay_flush(&mut self, delay_millis: u32) -> Result<(), IoError> {
+        let delay_tgt = delay_millis as u64;
+        let mf_sink = self.f_sink.clone();
+
+        if self.flush_time_tx.is_none() {
+            // no task running so start one
+            let (tx, rx) = async_channel::bounded(100);
+            self.flush_time_tx = Some(tx);
+            let delay_tgt = Duration::from_millis(delay_tgt);
+            let flush_count = self.flush_count.clone();
+
+            fluvio_future::task::spawn(async move {
+                let mut delay_dur = delay_tgt;
+                let mut write_time = Instant::now();
+
+                // when the mut_record struct is dropped self.flush_time_tx
+                // will also be closed and dropped, causing this while loop
+                // to end then exit the task
+                while !rx.is_closed() {
+                    if let Ok(wt) = rx.recv().await {
+                        // always grab inital write time, but also
+                        // guarantee a wait on a write time for flush
+                        // for wait times after the first delay runs
+                        write_time = wt;
+                    }
+                    //  A short fixed wait was added here to give a little
+                    //  breathing space to accumulate clusters of writes which
+                    //  arrive closely but not fast enough for this task loop
+                    //  between the first and following check for write times
+                    //  It prevents extra flushes and the time is still accounted for
+                    //  in the delay_dur calculation.
+                    timer::after(Duration::from_millis(3)).await;
```

Comment:
can you make 3 as constant?

Reply:
Yup made this a constant in the squash update